### PR TITLE
Add feature to filter submissions by team name in my submissions

### PR DESCRIPTION
**Pull request by Sanji515**
#### Changes proposed in this pull request:

- Added the input box to get the filter query as participant team name

- Used the challenge submission API with filter query for filtering the submissions

- Link to live demo: http://pr-166-evalai.surge.sh

### CODE REVIEW

**Comment by lunayach** [comment]
@Sanji515 resolve the conflicts and pass the build.

**Comment by RishabhJain2018** [comment]
@Sanji515 Please remove the conflicts here.

**Comment by Sanji515** [comment]
@RishabhJain2018 @lunayach @Shekharrajak Please review

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts [hunk 1]
```diff
@@ -209,9 +226,17 @@ export class ChallengesubmissionsComponent implements OnInit, AfterViewInit {
...
-    this.apiService.getUrl(API_PATH).subscribe(
+    let API_PATH;
+    if (SELF.filterSubmissionsQuery === '') {
+      API_PATH = SELF.endpointsService.challengeSubmissionURL(challenge, phase);
+    } else {
+      API_PATH = SELF.endpointsService.challengeSubmissionWithFilterQueryURL(
+        challenge, phase, SELF.filterSubmissionsQuery
+      );
+    }
+    console.log(API_PATH);
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts]: Please remove this.

File: src/app/components/utility/input/input.component.ts [hunk 2]
```diff
@@ -218,6 +218,10 @@ export class InputComponent implements OnInit {
        this.isValid = this.globalService.validatePassword(e);
        this.isValid ? this.message = '' : this.message = 'Password minimum 8 characters';
     }
+    if (this.name === 'bannedEmailIds' || this.name === 'filterByTeamName') {
```
Comment (lunayach) [inline src/app/components/utility/input/input.component.ts]: Is this change related to #163?

Comment (Sanji515) [reply src/app/components/utility/input/input.component.ts]: yeah actually this was build on top of that, therefore it is showing here. Ideally you should review that PR first.

**Review by lunayach — COMMENTED** [review]
It works! but only when the query string in the input search field matches **exactly** with some existing team name.  
A feature where it fetches all the Teams having their name starting (and/or some other criteria apart from the **exact** one) with what's already typed in the search input field might be more useful. 

Should we open an issue for that @RishabhJain2018? (cc: @Sanji515 )

File: src/app/components/utility/input/input.component.ts (hunk 2 above)
Comment (lunayach) [reply src/app/components/utility/input/input.component.ts]: Okay, cool.

Comment (Sanji515) [reply src/app/components/utility/input/input.component.ts]: @lunayach Removed it from here for now, it will get added in `ban participant` PR

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts (hunk 1 above)
Comment (Sanji515) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts]: done

**Review by lunayach — APPROVED** [review]
LGTM! (Just one comment https://github.com/Cloud-CV/EvalAI-ngx/pull/166#pullrequestreview-264506364)

**Comment by RishabhJain2018** [comment]
> A feature where it fetches all the Teams having their name starting (and/or some other criteria apart from the exact one) with what's already typed in the search input field might be more useful.

Yes, @Sanji515 Please open up an issue for this as well.

**Comment by Sanji515** [comment]
@RishabhJain2018 as it requires the backend change only so instead of issue how about if I open a PR in `EvalAI` for filtering even it matches the starting keywords of the query.

**Comment by RishabhJain2018** [comment]
Please open it there and mention this PR as a reference.
